### PR TITLE
[SecurityBundle] use libsodium to run Argon2i related tests

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26038
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -143,13 +143,19 @@ before_install:
           (cd php-$MIN_PHP && ./configure --enable-sigchild --enable-pcntl && make -j2)
       fi
 
+    # install libsodium
+    - sudo add-apt-repository ppa:ondrej/php -y
+    - sudo apt-get update -q
+    - sudo apt-get install libsodium-dev -y
```
Comment (nicolas-grekas) [inline .travis.yml]: I'd suggest to move these 3 lines below `Install extra PHP extensions`, in the same block (no dashes before)

Comment (nicolas-grekas) [reply .travis.yml]: oh, and move them in the elif below, and wrap them in another if that checks if the `sodium.so` exists in the cache already or not

**Review by chalasr — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -149,7 +149,13 @@ before_install:
           ([[ $deps ]] || tfold ext.symfony_debug 'cd src/Symfony/Component/Debug/Resources/ext && phpize && ./configure && make && echo extension = $(pwd)/modules/symfony_debug.so >> '"$INI")
           tfold ext.apcu tpecl apcu-4.0.11 apcu.so
       elif [[ ! $skip && $PHP = 7.* ]]; then
+          # install libsodium
```
Comment (nicolas-grekas) [inline .travis.yml]: `if [[ ! -e ~/php-ext/$(php -r "echo basename(ini_get('extension_dir'));")/libsodium/sodium.so ]]; then`

Comment (xabbuh) [reply .travis.yml]: done

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**